tests/code-splitting/subtests/dynamic-import-preloading/parcel/index.md

```diff
@@ -0,0 +1,5 @@
+---
+result: fail
```

Comment:
We usually don’t allow a test to be marked as a `fail` without having opened an issue on the tool’s issue tracker. Please open an issue with Parcel and link to it here (my hunch is their plugin infrastructure will be able to solve this once documented).

I also think the explanation here is very vague. Maybe you can re-word this a bit? Something like “Parcel does not automatically preload a module’s dependencies nor does it provide access to the module graph which is required to determine what needs to be preloaded” or something.

Reply:
@surma Challenge with Parcel is that the module's dependency is already there in the parent module. So whenever our on-demand module loads, its dependency is already in the parent module, so no network request is triggered.
`index.js(Already has the exclaim.js)` -> `utilsA.js`

I am confused as to what to mention in the issue. Should I request them to create another bundle for `exclaim.js` and load it in parallel?

Reply:
You can put this into package.json to force Parcel to create a shared bundle in this case:

```json
{
	"@parcel/bundler-default": {
		"minBundleSize": 0
	}
}
```
With that, there is no waterfall:
<img width="853" alt="Bildschirmfoto 2020-07-25 um 11 58 39" src="https://user-images.githubusercontent.com/4586894/88454206-2fafdb80-ce6e-11ea-8d4e-b6235dc2b5fd.png">

Reply:
Is this config option documented somewhere? If so, there’s a couple of tests where we can remove `const reallyLongString` now :D

Reply:
Soon™️: https://github.com/parcel-bundler/website/pull/636

Reply:
@mischnic Thanks a lot!!!! After making `exclaim.js` bigger than 20 KiB, I am able to simulate the test conditions. And I am updating the test results also, because parcel is passing it.

Reply:
> You can put this into package.json to force Parcel to create a shared bundle in this case (or make exclaim.js at least 20kb big):
> 
> ```json
> {
> 	"@parcel/bundler-default": {
> 		"minBundleSize": 0
> 	}
> }
> ```
> With that, there is no waterfall:
> <img width="853" alt="Bildschirmfoto 2020-07-25 um 11 58 39" src="https://user-images.githubusercontent.com/4586894/88454206-2fafdb80-ce6e-11ea-8d4e-b6235dc2b5fd.png">
> 

This screenshot looks odd to me. Shouldn't it only load utilsA or utilsB randomly? Not both. And shouldn't there be an additional request to the file with the shared depenency?

Reply:
You can't trust the naming filenames here,

`index.xxxxxx.js` is the main JS bundle, `utilsA.xxxxxx.js` is the shared bundle with `exclaim.js`, `utilsB.xxxx.js` is the `utilsB.js` child bundle:

```
$ parcel build index.html --detailed-report
✨ Built in 4.32s

dist/index.6d2e9075.js                                  354 B    1.49s
├── node_modules/@parcel/runtime-js/lib/JSRuntime.js    266 B     52ms
└── index.js                                             88 B      0ms

dist/utilsA.b049957f.js                                 226 B    1.49s
├── Code from unknown sourcefiles                       157 B      0ms
└── utilsA.js                                            69 B     54ms

dist/utilsB.8e0c1533.js                                 216 B    1.49s
├── Code from unknown sourcefiles                       157 B      0ms
└── utilsB.js                                            59 B     54ms

dist/utilsA.231952d4.js                                 151 B    1.49s
├── exclaim.js                                           88 B     20ms
└── Code from unknown sourcefiles                        63 B      0ms

dist/index.html                                          56 B    1.48s
└── index.html                                           71 B    110ms
✨  Done in 5.67s.
```